src/processes/follow-topic-process.js

```diff
@@ -191,16 +191,22 @@ const followTopicProcess = async (job, done) => {
         return;
       }
 
+      // use dummy sender user id if provided (for testing purpose)
+      const senderUserId = data?.dummy_sender_user_id || communityMessageFormat.user_id;
       // check if user is anonymous
-      const [senderUser] = await Promise.all([User.findByPk(communityMessageFormat.user_id)]);
+      const [senderUser] = await Promise.all([User.findByPk(senderUserId)]);
+      const communityMessageFormatObject = {
+        user_id: senderUserId,
+        message: communityMessageFormat.message
+      };
 
       if (senderUser.is_anonymous) {
-        await sendMessageAsAnonymous(serverClient, communityMessageFormat, data);
+        await sendMessageAsAnonymous(serverClient, communityMessageFormatObject, data);
       } else {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code assumes that `senderUser` is not `null` or `undefined` and directly accesses its `is_anonymous` property. This could potentially cause a TypeError if `senderUser` is `null` or `undefined`. It would be safer to add a null check for `senderUser` before accessing its properties.
```diff
- if (senderUser.is_anonymous) {
+ if (senderUser && senderUser.is_anonymous) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->